# **데이터 전처리**

1. 데이터 불러오기, 합치기

In [ ]:
 import pandas as pd
 import numpy as np
 import warnings; warnings.filterwarnings('ignore')

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


train : 2022년 9월 ~ 2023년 8월 버스 데이터

test : 2023년 9월, 10월, 11월 버스 데이터

In [ ]:
#월별 버스 데이터를 train 하나로 합치기
path = '/content/drive/MyDrive/Bus_Tayo/bus'
train_files = ['2022_09.csv', '2022_10.csv', '2022_11.csv', '2022_12.csv', '2023_01.csv', '2023_02.csv',
        '2023_03.csv', '2023_04.csv', '2023_05.csv', '2023_06.csv', '2023_07.csv', '2023_08.csv']
train_df = [pd.read_csv(path+'/'+f) for f in train_files]
train = pd.concat(train_df)

In [ ]:
train.head()

사용년월 노선번호                    노선명  표준버스정류장ID 버스정류장ARS번호  \
0  202209  741      741번(진관차고지~헌인릉입구)  100000001       1001   
1  202209  470       470번(상암차고지~안골마을)  100000001       1001   
2  202209  N37  N37번(송파공영차고지~진관공영차고지)  100000001       1001   
3  202209  N37  N37번(진관공영차고지~송파공영차고지)  100000001       1001   
4  202209  100         100번(하계동~용산구청)  100000002       1002   

                   역명  00시승차총승객수  00시하차총승객수  1시승차총승객수  1시하차총승객수  ...  \
0      종로2가사거리(00077)         92         77       105       118  ...   
1      종로2가사거리(00067)         74         74         3         5  ...   
2      종로2가사거리(00032)          1          5       287       208  ...   
3      종로2가사거리(00089)          0          0         0         0  ...   
4  창경궁.서울대학교병원(00031)          0          0         0         0  ...   

   20시하차총승객수  21시승차총승객수  21시하차총승객수  22시승차총승객수  22시하차총승객수  23시승차총승객수  \
0        194        263        179        243        163        214   
1        185        342        169        256        164        231   
2          0          0          0          0          0          0   
3          0          0          0          0          0          0   
4         23         55         36         17         23         27   

   23시하차총승객수  교통수단타입코드  교통수단타입명      등록일자  
0        135        10   서울간선버스  20221003  
1        131        10   서울간선버스  20221003  
2          0        51   서울심야버스  20221003  
3          0        51   서울심야버스  20221003  
4          5        10   서울간선버스  20221003  

[5 rows x 57 columns]

In [ ]:
train.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 498176 entries, 0 to 41649
Data columns (total 57 columns):
 #   Column      Non-Null Count   Dtype 
---  ------      --------------   ----- 
 0   사용년월        498176 non-null  int64 
 1   노선번호        498176 non-null  object
 2   노선명         498176 non-null  object
 3   표준버스정류장ID   498176 non-null  int64 
 4   버스정류장ARS번호  498176 non-null  object
 5   역명          498176 non-null  object
 6   00시승차총승객수   498176 non-null  int64 
 7   00시하차총승객수   498176 non-null  int64 
 8   1시승차총승객수    498176 non-null  int64 
 9   1시하차총승객수    498176 non-null  int64 
 10  2시승차총승객수    498176 non-null  int64 
 11  2시하차총승객수    498176 non-null  int64 
 12  3시승차총승객수    498176 non-null  int64 
 13  3시하차총승객수    498176 non-null  int64 
 14  4시승차총승객수    498176 non-null  int64 
 15  4시하차총승객수    498176 non-null  int64 
 16  5시승차총승객수    498176 non-null  int64 
 17  5시하차총승객수    498176 non-null  int64 
 18  6시승차총승객수    498176 non-null  int64 
 19  6시하차총승객수    498176 non-n

2. 불필요한 피처 삭제

등록일자, 교통수단타입코드, 버스정류장ARS번호, 노선명 칼럼을 삭제한다.

In [ ]:
train = train.drop(['등록일자', '교통수단타입코드', '버스정류장ARS번호', '노선명'], axis=1)
train.head()

사용년월 노선번호  표준버스정류장ID                  역명  00시승차총승객수  00시하차총승객수  1시승차총승객수  \
0  202209  741  100000001      종로2가사거리(00077)         92         77       105   
1  202209  470  100000001      종로2가사거리(00067)         74         74         3   
2  202209  N37  100000001      종로2가사거리(00032)          1          5       287   
3  202209  N37  100000001      종로2가사거리(00089)          0          0         0   
4  202209  100  100000002  창경궁.서울대학교병원(00031)          0          0         0   

   1시하차총승객수  2시승차총승객수  2시하차총승객수  ...  19시하차총승객수  20시승차총승객수  20시하차총승객수  \
0       118         0         0  ...        389        301        194   
1         5         0         0  ...        386        327        185   
2       208       116        91  ...          0          0          0   
3         0        77        74  ...          0          0          0   
4         0         0         0  ...         60        235         23   

   21시승차총승객수  21시하차총승객수  22시승차총승객수  22시하차총승객수  23시승차총승객수  23시하차총승객수  교통수단타입명  
0        263        179        243        163        214        135   서울간선버스  
1        342        169        256        164        231        131   서울간선버스  
2          0          0          0          0          0          0   서울심야버스  
3          0          0          0          0          0          0   서울심야버스  
4         55         36         17         23         27          5   서울간선버스  

[5 rows x 53 columns]

3. 심야버스 삭제

출퇴근 시간대를 분석할 예정이므로 심야버스 row를 삭제한다.

In [ ]:
train = train[train['교통수단타입명'] != '서울심야버스']
count = train[train['교통수단타입명'] == '서울심야버스'].shape[0]
print(f'서울심야버스인 행의 개수: {count}')

서울심야버스인 행의 개수: 0


4. 출퇴근 시간대를 제외한 시간대 column 삭제, 2시간 단위로 시간대 재구성

출퇴근 시간이 아닌 20시~5시 데이터를 삭제한다.

6-8, 8-10, 10-12, 12-14, 14-16, 16-18, 18-20, 20-22로 2시간 단위로 시간대를 묶는다.

X~Y ride(2시간 단위 승차 총 승객 수), X~Y takeoff(2시간 단위 하차 총 승객 수)로 칼럼명을 변경한다.

In [ ]:
train['6~8 ride'] = train['6시승차총승객수'] + train['7시승차총승객수']
train['8~10 ride'] = train['8시승차총승객수'] + train['9시승차총승객수']
train['10~12 ride'] = train['10시승차총승객수'] + train['11시승차총승객수']
train['12~14 ride'] = train['12시승차총승객수'] + train['13시승차총승객수']
train['14~16 ride'] = train['14시승차총승객수'] + train['15시승차총승객수']
train['16~18 ride'] = train['16시승차총승객수'] + train['17시승차총승객수']
train['18~20 ride'] = train['18시승차총승객수'] + train['19시승차총승객수']

time_drop_columns = [f'{hour}시승차총승객수' for hour in range(1,24)]
train.drop(time_drop_columns, axis=1, inplace=True)
train.drop('00시승차총승객수', axis=1, inplace=True)

train['6~8 takeoff'] = train['6시하차총승객수'] + train['7시하차총승객수']
train['8~10 takeoff'] = train['8시하차총승객수'] + train['9시하차총승객수']
train['10~12 takeoff'] = train['10시하차총승객수'] + train['11시하차총승객수']
train['12~14 takeoff'] = train['12시하차총승객수'] + train['13시하차총승객수']
train['14~16 takeoff'] = train['14시하차총승객수'] + train['15시하차총승객수']
train['16~18 takeoff'] = train['16시하차총승객수'] + train['17시하차총승객수']
train['18~20 takeoff'] = train['18시하차총승객수'] + train['19시하차총승객수']

time_drop_columns = [f'{hour}시하차총승객수' for hour in range(1,24)]
train.drop(time_drop_columns, axis=1, inplace=True)
train.drop('00시하차총승객수', axis=1, inplace=True)

In [ ]:
train.head()

사용년월 노선번호  표준버스정류장ID                  역명 교통수단타입명  6~8 ride  8~10 ride  \
0  202209  741  100000001      종로2가사거리(00077)  서울간선버스       654        809   
1  202209  470  100000001      종로2가사거리(00067)  서울간선버스       788       1265   
4  202209  100  100000002  창경궁.서울대학교병원(00031)  서울간선버스       227        677   
5  202209  107  100000002  창경궁.서울대학교병원(00058)  서울간선버스        89        171   
6  202209  104  100000002  창경궁.서울대학교병원(00024)  서울간선버스       376        759   

   10~12 ride  12~14 ride  14~16 ride  16~18 ride  18~20 ride  6~8 takeoff  \
0         576         574         643         651         758          354   
1        1217        1074         839         731         767          325   
4         282         193         381         558         374          708   
5         101          76          92          96         130          653   
6         381         312         344         464         393         1064   

   8~10 takeoff  10~12 takeoff  12~14 takeoff  14~16 takeoff  16~18 takeoff  \
0           753            769            800            763            991   
1           782            696            982            803            950   
4          1114            504            450            353            209   
5          1070            287            232            159             92   
6          1765            509            460            451            228   

   18~20 takeoff  
0           1016  
1            903  
4            146  
5             62  
6            148

5. 역명 칼럼 변환 및 칼럼 이름 재구성

역명 칼럼을 괄호 안 5자리 숫자(00000)로 대체한다. 또한, 이름을 '역번호'로 수정한다.

사용년월, 노선번호, 역명, 표준버스정류장ID, 교통수단타입명, 역번호를 영문으로 변경한다.

In [ ]:
train['역번호'] = train['역명'].str.extract(r'\((\d+)\)')
train.drop('역명', axis=1, inplace=True)
train.head()

사용년월 노선번호  표준버스정류장ID 교통수단타입명  6~8 ride  8~10 ride  10~12 ride  \
0  202209  741  100000001  서울간선버스       654        809         576   
1  202209  470  100000001  서울간선버스       788       1265        1217   
4  202209  100  100000002  서울간선버스       227        677         282   
5  202209  107  100000002  서울간선버스        89        171         101   
6  202209  104  100000002  서울간선버스       376        759         381   

   12~14 ride  14~16 ride  16~18 ride  18~20 ride  6~8 takeoff  8~10 takeoff  \
0         574         643         651         758          354           753   
1        1074         839         731         767          325           782   
4         193         381         558         374          708          1114   
5          76          92          96         130          653          1070   
6         312         344         464         393         1064          1765   

   10~12 takeoff  12~14 takeoff  14~16 takeoff  16~18 takeoff  18~20 takeoff  \
0            769            800            763            991           1016   
1            696            982            803            950            903   
4            504            450            353            209            146   
5            287            232            159             92             62   
6            509            460            451            228            148   

     역번호  
0  00077  
1  00067  
4  00031  
5  00058  
6  00024

In [ ]:
columns={'사용년월':'month', '노선번호': 'bus_ID', '표준버스정류장ID': 'bus_station_ID',
        '교통수단타입명':'type', '역번호': 'bus_station_ID', '표준버스정류장ID': 'bus_station_SID'}

train.rename(columns=columns, inplace=True)

train.head()

month bus_ID  bus_station_SID    type  6~8 ride  8~10 ride  10~12 ride  \
0  202209    741        100000001  서울간선버스       654        809         576   
1  202209    470        100000001  서울간선버스       788       1265        1217   
4  202209    100        100000002  서울간선버스       227        677         282   
5  202209    107        100000002  서울간선버스        89        171         101   
6  202209    104        100000002  서울간선버스       376        759         381   

   12~14 ride  14~16 ride  16~18 ride  18~20 ride  6~8 takeoff  8~10 takeoff  \
0         574         643         651         758          354           753   
1        1074         839         731         767          325           782   
4         193         381         558         374          708          1114   
5          76          92          96         130          653          1070   
6         312         344         464         393         1064          1765   

   10~12 takeoff  12~14 takeoff  14~16 takeoff  16~18 takeoff  18~20 takeoff  \
0            769            800            763            991           1016   
1            696            982            803            950            903   
4            504            450            353            209            146   
5            287            232            159             92             62   
6            509            460            451            228            148   

  bus_station_ID  
0          00077  
1          00067  
4          00031  
5          00058  
6          00024

6. Object type 피처 처리

bus_ID는 110A, 종로09와 같이 숫자와 문자열이 함께 섞여 있다.

따라서, 숫자형으로 변환하기 쉽도록 bus_ID 대신 노선 정보 데이터 속 route_ID로 대체한다.

노선 정보 데이터 링크 : https://data.seoul.go.kr/dataList/OA-15262/S/1/datasetView.do

In [ ]:
#노선 정보 데이터 불러오기
path = '/content/drive/MyDrive/Bus_Tayo/data'
route_info = pd.read_csv(path+'/route_info.csv', encoding='utf-8')
route_info.rename(columns={'노선명': 'bus_ID', 'ROUTEID': 'route_ID'}, inplace=True)
route_info

bus_ID   route_ID
0     0017  100100124
1      01A  100100001
2      01B  106000004
3     0411  104000012
4      100  100100549
..     ...        ...
671   종로11  100900007
672   종로12  100900009
673   종로13  100900002
674   중랑01  106900001
675   중랑02  106900002

[676 rows x 2 columns]

In [ ]:
#bus_ID를 기준으로 하여 train과 route_info 합치기
train = pd.merge(train, route_info, on='bus_ID', how='left')

train

month bus_ID  bus_station_SID    type  6~8 ride  8~10 ride  \
0       202209    741        100000001  서울간선버스       654        809   
1       202209    470        100000001  서울간선버스       788       1265   
2       202209    100        100000002  서울간선버스       227        677   
3       202209    107        100000002  서울간선버스        89        171   
4       202209    104        100000002  서울간선버스       376        759   
...        ...    ...              ...     ...       ...        ...   
450525  202308    672        999800002  서울간선버스         0          0   
450526  202308   9404        999800003  서울광역버스         2          2   
450527  202308   9404        999800004  서울광역버스         0          0   
450528  202308   9707        999800005  서울광역버스         0          0   
450529  202308   9701        999800005  서울광역버스         0          0   

        10~12 ride  12~14 ride  14~16 ride  16~18 ride  18~20 ride  \
0              576         574         643         651         758   
1             1217        1074         839         731         767   
2              282         193         381         558         374   
3              101          76          92          96         130   
4              381         312         344         464         393   
...            ...         ...         ...         ...         ...   
450525           0           0           0           0           0   
450526           0           1           1           1           2   
450527           0           0           0           0           0   
450528           0           0           0           0           0   
450529           0           0           0           0           0   

        6~8 takeoff  8~10 takeoff  10~12 takeoff  12~14 takeoff  \
0               354           753            769            800   
1               325           782            696            982   
2               708          1114            504            450   
3               653          1070            287            232   
4              1064          1765            509            460   
...             ...           ...            ...            ...   
450525            0             1              0              0   
450526            7             0              1              1   
450527            1             3              1              2   
450528            2            11              3              7   
450529            0            25              3              1   

        14~16 takeoff  16~18 takeoff  18~20 takeoff bus_station_ID  \
0                 763            991           1016          00077   
1                 803            950            903          00067   
2                 353            209            146          00031   
3                 159             92             62          00058   
4                 451            228            148          00024   
...               ...            ...            ...            ...   
450525              0              1              0          00081   
450526              1              3              0          00001   
450527              5              3              8          00060   
450528              2             11              9          00078   
450529              0              2              1          00127   

           route_ID  
0       123000010.0  
1       100100073.0  
2       100100549.0  
3       100100012.0  
4       100100009.0  
...             ...  
450525  100100495.0  
450526  100100391.0  
450527  100100391.0  
450528  100100400.0  
450529  100100397.0  

[450530 rows x 20 columns]

In [ ]:
#route_ID 값이 없는 bus_ID가 존재하는지 확인
null_routes_df = train[train['route_ID'].isnull()][['bus_ID']].drop_duplicates()
print("route_ID 값이 null인 bus_ID:")
print(null_routes_df)

route_ID 값이 null인 bus_ID:
        bus_ID
37           1
174       9703
204        153
540       9403
1989      성동07
2006        17
2236       411
3587      성동06
5412      2235
6048      2234
7274      8112
7655      1156
7661      노원04
8227    성북10-2
8229    성북10-1
13848     은평01
16224     8761
16940     마포14
17538     6649
20190     6513
20318     6611
21498     6635
22871     6654
59772   금천01-1
188629  Jan.01
262874    8001


버스 노선 고유 아이디(route_ID)와 매핑되지 못한 데이터가 존재한다.

수동으로 route_ID를 찾을 수 있는 경우 직접 찾아 넣었다.

매핑되지 못하며 수동으로 route_ID를 찾을 수 없는 버스는 임의로 route_ID를 부여해 처리하였다.
111110000~111110011 사이 숫자로 부여하였다.

In [ ]:
train[train['route_ID'].isnull()]['bus_ID'].count()

9863

In [ ]:
#route_ID가 없는 bus_ID 처리
#검색을 통해 route_ID를 찾을 수 있는 경우 직접 찾아서 넣음
#그렇지 못한 경우 111110000~111110011 사이 숫자를 임의로 부여하여 처리함
null_routes = {'1': 100100001, '9703': 100100398, '153': 111110000, '9403': 100100390, '성동07': 103900005,
               '성동06': 103900007, '2235': 111110001, '2234': 111110002, '8112': 111110003, '1156': 100100166,
               '1157': 100100166, '노원04': 111110004, '성북10-2': 107900017, '성북10-1': 107900014,
               '은평01': 111900002, '8761': 111110005, '마포14': 113900006, '6649': 114000001,
               '6513': 111110007, '6611': 100100295, '6635': 100100311, '6654': 100100322,
                '금천01-1': 117900008, '8001': 106000004, '17': 111110008, '411': 111110009, '6511': 111110010,
               'Jan.01': 111110011}
null_routes = pd.DataFrame.from_dict(null_routes, orient='index', columns=['route_ID']).reset_index()
null_routes.columns = ['bus_ID', 'route_ID']

null_routes

bus_ID   route_ID
0        1  100100001
1     9703  100100398
2      153  111110000
3     9403  100100390
4     성동07  103900005
5     성동06  103900007
6     2235  111110001
7     2234  111110002
8     8112  111110003
9     1156  100100166
10    1157  100100166
11    노원04  111110004
12  성북10-2  107900017
13  성북10-1  107900014
14    은평01  111900002
15    8761  111110005
16    마포14  113900006
17    6649  114000001
18    6513  111110007
19    6611  100100295
20    6635  100100311
21    6654  100100322
22  금천01-1  117900008
23    8001  106000004
24      17  111110008
25     411  111110009
26    6511  111110010
27  Jan.01  111110011

In [ ]:
train = pd.merge(train, null_routes, on='bus_ID', how='left')
train


month bus_ID  bus_station_SID    type  6~8 ride  8~10 ride  \
0       202209    741        100000001  서울간선버스       654        809   
1       202209    470        100000001  서울간선버스       788       1265   
2       202209    100        100000002  서울간선버스       227        677   
3       202209    107        100000002  서울간선버스        89        171   
4       202209    104        100000002  서울간선버스       376        759   
...        ...    ...              ...     ...       ...        ...   
450525  202308    672        999800002  서울간선버스         0          0   
450526  202308   9404        999800003  서울광역버스         2          2   
450527  202308   9404        999800004  서울광역버스         0          0   
450528  202308   9707        999800005  서울광역버스         0          0   
450529  202308   9701        999800005  서울광역버스         0          0   

        10~12 ride  12~14 ride  14~16 ride  16~18 ride  ...  6~8 takeoff  \
0              576         574         643         651  ...          354   
1             1217        1074         839         731  ...          325   
2              282         193         381         558  ...          708   
3              101          76          92          96  ...          653   
4              381         312         344         464  ...         1064   
...            ...         ...         ...         ...  ...          ...   
450525           0           0           0           0  ...            0   
450526           0           1           1           1  ...            7   
450527           0           0           0           0  ...            1   
450528           0           0           0           0  ...            2   
450529           0           0           0           0  ...            0   

        8~10 takeoff  10~12 takeoff  12~14 takeoff  14~16 takeoff  \
0                753            769            800            763   
1                782            696            982            803   
2               1114            504            450            353   
3               1070            287            232            159   
4               1765            509            460            451   
...              ...            ...            ...            ...   
450525             1              0              0              0   
450526             0              1              1              1   
450527             3              1              2              5   
450528            11              3              7              2   
450529            25              3              1              0   

        16~18 takeoff  18~20 takeoff  bus_station_ID   route_ID_x  route_ID_y  
0                 991           1016           00077  123000010.0         NaN  
1                 950            903           00067  100100073.0         NaN  
2                 209            146           00031  100100549.0         NaN  
3                  92             62           00058  100100012.0         NaN  
4                 228            148           00024  100100009.0         NaN  
...               ...            ...             ...          ...         ...  
450525              1              0           00081  100100495.0         NaN  
450526              3              0           00001  100100391.0         NaN  
450527              3              8           00060  100100391.0         NaN  
450528             11              9           00078  100100400.0         NaN  
450529              2              1           00127  100100397.0         NaN  

[450530 rows x 21 columns]

In [ ]:
#필요하지 않은 route_ID_x와 route_ID_y 칼럼 삭제
train['route_ID'] = train['route_ID_x'].fillna(train['route_ID_y'])
train.drop(['route_ID_x', 'route_ID_y'], axis=1, inplace=True)
train.insert(4, 'route_ID', train.pop('route_ID'))

In [ ]:
train.head()

month bus_ID  bus_station_SID    type     route_ID  6~8 ride  8~10 ride  \
0  202209    741        100000001  서울간선버스  123000010.0       654        809   
1  202209    470        100000001  서울간선버스  100100073.0       788       1265   
2  202209    100        100000002  서울간선버스  100100549.0       227        677   
3  202209    107        100000002  서울간선버스  100100012.0        89        171   
4  202209    104        100000002  서울간선버스  100100009.0       376        759   

   10~12 ride  12~14 ride  14~16 ride  16~18 ride  18~20 ride  6~8 takeoff  \
0         576         574         643         651         758          354   
1        1217        1074         839         731         767          325   
2         282         193         381         558         374          708   
3         101          76          92          96         130          653   
4         381         312         344         464         393         1064   

   8~10 takeoff  10~12 takeoff  12~14 takeoff  14~16 takeoff  16~18 takeoff  \
0           753            769            800            763            991   
1           782            696            982            803            950   
2          1114            504            450            353            209   
3          1070            287            232            159             92   
4          1765            509            460            451            228   

   18~20 takeoff bus_station_ID  
0           1016          00077  
1            903          00067  
2            146          00031  
3             62          00058  
4            148          00024

In [ ]:
null_routes_df = train[train['route_ID'].isnull()][['bus_ID']]
print("route_ID 값이 null인 bus_ID:")
print(null_routes_df)
train.drop(['bus_ID'], axis=1, inplace=True)

route_ID 값이 null인 bus_ID:
Empty DataFrame
Columns: [bus_ID]
Index: []


route_ID와 bus_ID 전부 매핑되었다.

이어서 bus_station_ID와 bus_station_SID를 연결하여 하나의 고유키를 생성한다.
예) bus_station_ID가 00077, bus_station_SID가 100000001인 경우 bus_station_PK는 00077100000001이 된다.

In [ ]:
#고유키 생성 및 기존 칼럼 삭제
train['bus_station_PK'] = train['bus_station_ID'].astype(str) + train['bus_station_SID'].astype(str)
train.drop(['bus_station_ID'], axis=1, inplace=True)
train.drop(['bus_station_SID'], axis=1, inplace=True)
#데이터 자료형을 정수형으로 바꿈
train['bus_station_PK'] = train['bus_station_PK'].astype(int)
train.head()

month    type     route_ID  6~8 ride  8~10 ride  10~12 ride  12~14 ride  \
0  202209  서울간선버스  123000010.0       654        809         576         574   
1  202209  서울간선버스  100100073.0       788       1265        1217        1074   
2  202209  서울간선버스  100100549.0       227        677         282         193   
3  202209  서울간선버스  100100012.0        89        171         101          76   
4  202209  서울간선버스  100100009.0       376        759         381         312   

   14~16 ride  16~18 ride  18~20 ride  6~8 takeoff  8~10 takeoff  \
0         643         651         758          354           753   
1         839         731         767          325           782   
2         381         558         374          708          1114   
3          92          96         130          653          1070   
4         344         464         393         1064          1765   

   10~12 takeoff  12~14 takeoff  14~16 takeoff  16~18 takeoff  18~20 takeoff  \
0            769            800            763            991           1016   
1            696            982            803            950            903   
2            504            450            353            209            146   
3            287            232            159             92             62   
4            509            460            451            228            148   

   bus_station_PK  
0     77100000001  
1     67100000001  
2     31100000002  
3     58100000002  
4     24100000002

In [ ]:
#칼럼별 자료형 확인
train.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 450530 entries, 0 to 450529
Data columns (total 18 columns):
 #   Column          Non-Null Count   Dtype  
---  ------          --------------   -----  
 0   month           450530 non-null  int64  
 1   type            450530 non-null  object 
 2   route_ID        450530 non-null  float64
 3   6~8 ride        450530 non-null  int64  
 4   8~10 ride       450530 non-null  int64  
 5   10~12 ride      450530 non-null  int64  
 6   12~14 ride      450530 non-null  int64  
 7   14~16 ride      450530 non-null  int64  
 8   16~18 ride      450530 non-null  int64  
 9   18~20 ride      450530 non-null  int64  
 10  6~8 takeoff     450530 non-null  int64  
 11  8~10 takeoff    450530 non-null  int64  
 12  10~12 takeoff   450530 non-null  int64  
 13  12~14 takeoff   450530 non-null  int64  
 14  14~16 takeoff   450530 non-null  int64  
 15  16~18 takeoff   450530 non-null  int64  
 16  18~20 takeoff   450530 non-null  int64  
 17  bus_statio

버스 타입 카테고리를 다음과 같은 숫자로 바꾼다.

type

서울간선버스(0), 서울지선버스(1), 서울마을버스(2), 서울순환버스(3), 서울광역버스(4)

In [ ]:
type_category = {
    '서울간선버스': 0,
    '서울지선버스': 1,
    '서울마을버스': 2,
    '서울순환버스': 3,
    '서울광역버스': 4
}
train['type'].replace(type_category, inplace=True)

train.head()

month  type     route_ID  6~8 ride  8~10 ride  10~12 ride  12~14 ride  \
0  202209     0  123000010.0       654        809         576         574   
1  202209     0  100100073.0       788       1265        1217        1074   
2  202209     0  100100549.0       227        677         282         193   
3  202209     0  100100012.0        89        171         101          76   
4  202209     0  100100009.0       376        759         381         312   

   14~16 ride  16~18 ride  18~20 ride  6~8 takeoff  8~10 takeoff  \
0         643         651         758          354           753   
1         839         731         767          325           782   
2         381         558         374          708          1114   
3          92          96         130          653          1070   
4         344         464         393         1064          1765   

   10~12 takeoff  12~14 takeoff  14~16 takeoff  16~18 takeoff  18~20 takeoff  \
0            769            800            763            991           1016   
1            696            982            803            950            903   
2            504            450            353            209            146   
3            287            232            159             92             62   
4            509            460            451            228            148   

   bus_station_PK  
0     77100000001  
1     67100000001  
2     31100000002  
3     58100000002  
4     24100000002

앞서 train에 적용한 과정을 똑같이 test에 적용한다.

In [ ]:
#월별 버스 데이터를 train 하나로 합치기
path = '/content/drive/MyDrive/Bus_Tayo/bus'
test_files = ['2023_09.csv', '2023_10.csv', '2023_11.csv']
test_df = [pd.read_csv(path+'/'+f) for f in test_files]
test = pd.concat(test_df)

In [ ]:
test.head()

사용년월 노선번호                    노선명  표준버스정류장ID 버스정류장ARS번호  \
0  202309  N37  N37번(송파공영차고지~진관공영차고지)  100000001       1001   
1  202309  470       470번(상암차고지~안골마을)  100000001       1001   
2  202309  N37  N37번(진관공영차고지~송파공영차고지)  100000001       1001   
3  202309  741      741번(진관차고지~헌인릉입구)  100000001       1001   
4  202309  100         100번(하계동~용산구청)  100000002       1002   

                   역명  00시승차총승객수  00시하차총승객수  1시승차총승객수  1시하차총승객수  ...  \
0      종로2가사거리(00032)         11         18       168       236  ...   
1      종로2가사거리(00066)         83        109         7         3  ...   
2      종로2가사거리(00089)          0          0         0         0  ...   
3      종로2가사거리(00075)        118        122       119       160  ...   
4  창경궁.서울대학교병원(00031)          0          0         0         0  ...   

   20시하차총승객수  21시승차총승객수  21시하차총승객수  22시승차총승객수  22시하차총승객수  23시승차총승객수  \
0          0          0          0          0          0          0   
1        221        419        260        365        246        261   
2          0          0          0          0          0          0   
3        279        303        231        227        196        223   
4         26         77         31         43         42         24   

   23시하차총승객수  교통수단타입코드  교통수단타입명      등록일자  
0          0        51   서울심야버스  20231003  
1        145        10   서울간선버스  20231003  
2          0        51   서울심야버스  20231003  
3        130        10   서울간선버스  20231003  
4          5        10   서울간선버스  20231003  

[5 rows x 57 columns]

In [ ]:
test.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 125576 entries, 0 to 41563
Data columns (total 57 columns):
 #   Column      Non-Null Count   Dtype 
---  ------      --------------   ----- 
 0   사용년월        125576 non-null  int64 
 1   노선번호        125576 non-null  object
 2   노선명         125576 non-null  object
 3   표준버스정류장ID   125576 non-null  int64 
 4   버스정류장ARS번호  125576 non-null  object
 5   역명          125576 non-null  object
 6   00시승차총승객수   125576 non-null  int64 
 7   00시하차총승객수   125576 non-null  int64 
 8   1시승차총승객수    125576 non-null  int64 
 9   1시하차총승객수    125576 non-null  int64 
 10  2시승차총승객수    125576 non-null  int64 
 11  2시하차총승객수    125576 non-null  int64 
 12  3시승차총승객수    125576 non-null  int64 
 13  3시하차총승객수    125576 non-null  int64 
 14  4시승차총승객수    125576 non-null  int64 
 15  4시하차총승객수    125576 non-null  int64 
 16  5시승차총승객수    125576 non-null  int64 
 17  5시하차총승객수    125576 non-null  int64 
 18  6시승차총승객수    125576 non-null  int64 
 19  6시하차총승객수    125576 non-n

In [ ]:
test = test.drop(['등록일자', '교통수단타입코드', '버스정류장ARS번호', '노선명'], axis=1)
test.head()

사용년월 노선번호  표준버스정류장ID                  역명  00시승차총승객수  00시하차총승객수  1시승차총승객수  \
0  202309  N37  100000001      종로2가사거리(00032)         11         18       168   
1  202309  470  100000001      종로2가사거리(00066)         83        109         7   
2  202309  N37  100000001      종로2가사거리(00089)          0          0         0   
3  202309  741  100000001      종로2가사거리(00075)        118        122       119   
4  202309  100  100000002  창경궁.서울대학교병원(00031)          0          0         0   

   1시하차총승객수  2시승차총승객수  2시하차총승객수  ...  19시하차총승객수  20시승차총승객수  20시하차총승객수  \
0       236       146       105  ...          0          0          0   
1         3         0         0  ...        447        378        221   
2         0       118        91  ...          0          0          0   
3       160         0         0  ...        512        292        279   
4         0         0         0  ...         36        152         26   

   21시승차총승객수  21시하차총승객수  22시승차총승객수  22시하차총승객수  23시승차총승객수  23시하차총승객수  교통수단타입명  
0          0          0          0          0          0          0   서울심야버스  
1        419        260        365        246        261        145   서울간선버스  
2          0          0          0          0          0          0   서울심야버스  
3        303        231        227        196        223        130   서울간선버스  
4         77         31         43         42         24          5   서울간선버스  

[5 rows x 53 columns]

In [ ]:
test = test[test['교통수단타입명'] != '서울심야버스']
count = test[test['교통수단타입명'] == '서울심야버스'].shape[0]
print(f'서울심야버스인 행의 개수: {count}')

서울심야버스인 행의 개수: 0


In [ ]:
test['6~8 ride'] = test['6시승차총승객수'] + test['7시승차총승객수']
test['8~10 ride'] = test['8시승차총승객수'] + test['9시승차총승객수']
test['10~12 ride'] = test['10시승차총승객수'] + test['11시승차총승객수']
test['12~14 ride'] = test['12시승차총승객수'] + test['13시승차총승객수']
test['14~16 ride'] = test['14시승차총승객수'] + test['15시승차총승객수']
test['16~18 ride'] = test['16시승차총승객수'] + test['17시승차총승객수']
test['18~20 ride'] = test['18시승차총승객수'] + test['19시승차총승객수']

time_drop_columns = [f'{hour}시승차총승객수' for hour in range(1,24)]
test.drop(time_drop_columns, axis=1, inplace=True)
test.drop('00시승차총승객수', axis=1, inplace=True)

test['6~8 takeoff'] = test['6시하차총승객수'] + test['7시하차총승객수']
test['8~10 takeoff'] = test['8시하차총승객수'] + test['9시하차총승객수']
test['10~12 takeoff'] = test['10시하차총승객수'] + test['11시하차총승객수']
test['12~14 takeoff'] = test['12시하차총승객수'] + test['13시하차총승객수']
test['14~16 takeoff'] = test['14시하차총승객수'] + test['15시하차총승객수']
test['16~18 takeoff'] = test['16시하차총승객수'] + test['17시하차총승객수']
test['18~20 takeoff'] = test['18시하차총승객수'] + test['19시하차총승객수']

time_drop_columns = [f'{hour}시하차총승객수' for hour in range(1,24)]
test.drop(time_drop_columns, axis=1, inplace=True)
test.drop('00시하차총승객수', axis=1, inplace=True)

In [ ]:
test['역번호'] = test['역명'].str.extract(r'\((\d+)\)')
test.drop('역명', axis=1, inplace=True)
test.head()

사용년월 노선번호  표준버스정류장ID 교통수단타입명  6~8 ride  8~10 ride  10~12 ride  \
1  202309  470  100000001  서울간선버스       699       1292        1191   
3  202309  741  100000001  서울간선버스       524        838         588   
4  202309  100  100000002  서울간선버스       258        546         313   
6  202309  104  100000002  서울간선버스       295        689         420   
8  202309  171  100000002  서울간선버스       578        911         651   

   12~14 ride  14~16 ride  16~18 ride  18~20 ride  6~8 takeoff  8~10 takeoff  \
1         988         744         672         720          349           716   
3         627         613         659         867          358           816   
4         256         334         482         365          681           977   
6         439         452         595         523          953          1377   
8         518         636         992         600          799          1682   

   10~12 takeoff  12~14 takeoff  14~16 takeoff  16~18 takeoff  18~20 takeoff  \
1            788            882            833            989           1028   
3            825            868            859            968           1200   
4            464            366            358            186            103   
6            575            471            388            161            138   
8            417            361            269            241            139   

     역번호  
1  00066  
3  00075  
4  00031  
6  00024  
8  00012

In [ ]:
columns={'사용년월':'month', '노선번호': 'bus_ID', '표준버스정류장ID': 'bus_station_ID',
        '교통수단타입명':'type', '역번호': 'bus_station_ID', '표준버스정류장ID': 'bus_station_SID'}

test.rename(columns=columns, inplace=True)

test.head()

month bus_ID  bus_station_SID    type  6~8 ride  8~10 ride  10~12 ride  \
1  202309    470        100000001  서울간선버스       699       1292        1191   
3  202309    741        100000001  서울간선버스       524        838         588   
4  202309    100        100000002  서울간선버스       258        546         313   
6  202309    104        100000002  서울간선버스       295        689         420   
8  202309    171        100000002  서울간선버스       578        911         651   

   12~14 ride  14~16 ride  16~18 ride  18~20 ride  6~8 takeoff  8~10 takeoff  \
1         988         744         672         720          349           716   
3         627         613         659         867          358           816   
4         256         334         482         365          681           977   
6         439         452         595         523          953          1377   
8         518         636         992         600          799          1682   

   10~12 takeoff  12~14 takeoff  14~16 takeoff  16~18 takeoff  18~20 takeoff  \
1            788            882            833            989           1028   
3            825            868            859            968           1200   
4            464            366            358            186            103   
6            575            471            388            161            138   
8            417            361            269            241            139   

  bus_station_ID  
1          00066  
3          00075  
4          00031  
6          00024  
8          00012

In [ ]:
path = '/content/drive/MyDrive/Bus_Tayo/data'
route_info = pd.read_csv(path+'/route_info.csv', encoding='utf-8')
route_info.rename(columns={'노선명': 'bus_ID', 'ROUTEID': 'route_ID'}, inplace=True)
route_info

bus_ID   route_ID
0     0017  100100124
1      01A  100100001
2      01B  106000004
3     0411  104000012
4      100  100100549
..     ...        ...
671   종로11  100900007
672   종로12  100900009
673   종로13  100900002
674   중랑01  106900001
675   중랑02  106900002

[676 rows x 2 columns]

In [ ]:
test = pd.merge(test, route_info, on='bus_ID', how='left')

test

month bus_ID  bus_station_SID    type  6~8 ride  8~10 ride  \
0       202309    470        100000001  서울간선버스       699       1292   
1       202309    741        100000001  서울간선버스       524        838   
2       202309    100        100000002  서울간선버스       258        546   
3       202309    104        100000002  서울간선버스       295        689   
4       202309    171        100000002  서울간선버스       578        911   
...        ...    ...              ...     ...       ...        ...   
113141  202311    672        999800002  서울간선버스         0          0   
113142  202311   9404        999800003  서울광역버스         3          2   
113143  202311   9404        999800004  서울광역버스         0          0   
113144  202311   9701        999800005  서울광역버스         0          0   
113145  202311   9707        999800005  서울광역버스         0          0   

        10~12 ride  12~14 ride  14~16 ride  16~18 ride  18~20 ride  \
0             1191         988         744         672         720   
1              588         627         613         659         867   
2              313         256         334         482         365   
3              420         439         452         595         523   
4              651         518         636         992         600   
...            ...         ...         ...         ...         ...   
113141           0           0           0           0           0   
113142           3           5           1           2           4   
113143           0           0           0           0           0   
113144           0           0           0           0           0   
113145           0           0           0           0           0   

        6~8 takeoff  8~10 takeoff  10~12 takeoff  12~14 takeoff  \
0               349           716            788            882   
1               358           816            825            868   
2               681           977            464            366   
3               953          1377            575            471   
4               799          1682            417            361   
...             ...           ...            ...            ...   
113141            0             0              1              0   
113142            7             1              0              2   
113143            5             1              1              0   
113144            0            19              4              2   
113145            0            22              5              8   

        14~16 takeoff  16~18 takeoff  18~20 takeoff bus_station_ID  \
0                 833            989           1028          00066   
1                 859            968           1200          00075   
2                 358            186            103          00031   
3                 388            161            138          00024   
4                 269            241            139          00012   
...               ...            ...            ...            ...   
113141              1              1              1          00081   
113142              0              2              5          00001   
113143              3              3              5          00060   
113144              3              6              5          00127   
113145              6              5              7          00078   

           route_ID  
0       100100073.0  
1       123000010.0  
2       100100549.0  
3       100100009.0  
4       100100036.0  
...             ...  
113141  100100495.0  
113142  100100391.0  
113143  100100391.0  
113144  100100397.0  
113145  100100400.0  

[113146 rows x 20 columns]

In [ ]:
#route_ID 값이 없는 bus_ID가 존재하는지 확인
null_routes_df = test[test['route_ID'].isnull()][['bus_ID']].drop_duplicates()
print("route_ID 값이 null인 bus_ID:")
print(null_routes_df)

route_ID 값이 null인 bus_ID:
       bus_ID
40          1
1592     8001
2045       17
2277      411
2415   Jan.01
4029     9403
13975    은평01
17151    마포14
17752    6649
23237    6654
78106   1.Jan


In [ ]:
test[test['route_ID'].isnull()]['bus_ID'].count()

1228

In [ ]:
#route_ID가 없는 bus_ID 처리
#검색을 통해 route_ID를 찾을 수 있는 경우 직접 찾아서 넣음
#그렇지 못한 경우 111110000~111110012 사이 숫자를 임의로 부여하여 처리함
null_routes = {'1': 100100001, '8001': 106000004, '17': 111110008, '411': 111110009,
               'Jan.01': 111110011, '9403': 100100390, '은평01': 111900002, '마포14': 113900006,
               '6649': 114000001, '6654': 100100322, '1.Jan': 111110012}
null_routes = pd.DataFrame.from_dict(null_routes, orient='index', columns=['route_ID']).reset_index()
null_routes.columns = ['bus_ID', 'route_ID']

null_routes

bus_ID   route_ID
0        1  100100001
1     8001  106000004
2       17  111110008
3      411  111110009
4   Jan.01  111110011
5     9403  100100390
6     은평01  111900002
7     마포14  113900006
8     6649  114000001
9     6654  100100322
10   1.Jan  111110012

In [ ]:
test = pd.merge(test, null_routes, on='bus_ID', how='left')
test

month bus_ID  bus_station_SID    type  6~8 ride  8~10 ride  \
0       202309    470        100000001  서울간선버스       699       1292   
1       202309    741        100000001  서울간선버스       524        838   
2       202309    100        100000002  서울간선버스       258        546   
3       202309    104        100000002  서울간선버스       295        689   
4       202309    171        100000002  서울간선버스       578        911   
...        ...    ...              ...     ...       ...        ...   
113141  202311    672        999800002  서울간선버스         0          0   
113142  202311   9404        999800003  서울광역버스         3          2   
113143  202311   9404        999800004  서울광역버스         0          0   
113144  202311   9701        999800005  서울광역버스         0          0   
113145  202311   9707        999800005  서울광역버스         0          0   

        10~12 ride  12~14 ride  14~16 ride  16~18 ride  ...  6~8 takeoff  \
0             1191         988         744         672  ...          349   
1              588         627         613         659  ...          358   
2              313         256         334         482  ...          681   
3              420         439         452         595  ...          953   
4              651         518         636         992  ...          799   
...            ...         ...         ...         ...  ...          ...   
113141           0           0           0           0  ...            0   
113142           3           5           1           2  ...            7   
113143           0           0           0           0  ...            5   
113144           0           0           0           0  ...            0   
113145           0           0           0           0  ...            0   

        8~10 takeoff  10~12 takeoff  12~14 takeoff  14~16 takeoff  \
0                716            788            882            833   
1                816            825            868            859   
2                977            464            366            358   
3               1377            575            471            388   
4               1682            417            361            269   
...              ...            ...            ...            ...   
113141             0              1              0              1   
113142             1              0              2              0   
113143             1              1              0              3   
113144            19              4              2              3   
113145            22              5              8              6   

        16~18 takeoff  18~20 takeoff  bus_station_ID   route_ID_x  route_ID_y  
0                 989           1028           00066  100100073.0         NaN  
1                 968           1200           00075  123000010.0         NaN  
2                 186            103           00031  100100549.0         NaN  
3                 161            138           00024  100100009.0         NaN  
4                 241            139           00012  100100036.0         NaN  
...               ...            ...             ...          ...         ...  
113141              1              1           00081  100100495.0         NaN  
113142              2              5           00001  100100391.0         NaN  
113143              3              5           00060  100100391.0         NaN  
113144              6              5           00127  100100397.0         NaN  
113145              5              7           00078  100100400.0         NaN  

[113146 rows x 21 columns]

In [ ]:
#필요하지 않은 route_ID_x와 route_ID_y 칼럼 삭제
test['route_ID'] = test['route_ID_x'].fillna(test['route_ID_y'])
test.drop(['route_ID_x', 'route_ID_y'], axis=1, inplace=True)
test.insert(4, 'route_ID', test.pop('route_ID'))

In [ ]:
test.head()

month bus_ID  bus_station_SID    type     route_ID  6~8 ride  8~10 ride  \
0  202309    470        100000001  서울간선버스  100100073.0       699       1292   
1  202309    741        100000001  서울간선버스  123000010.0       524        838   
2  202309    100        100000002  서울간선버스  100100549.0       258        546   
3  202309    104        100000002  서울간선버스  100100009.0       295        689   
4  202309    171        100000002  서울간선버스  100100036.0       578        911   

   10~12 ride  12~14 ride  14~16 ride  16~18 ride  18~20 ride  6~8 takeoff  \
0        1191         988         744         672         720          349   
1         588         627         613         659         867          358   
2         313         256         334         482         365          681   
3         420         439         452         595         523          953   
4         651         518         636         992         600          799   

   8~10 takeoff  10~12 takeoff  12~14 takeoff  14~16 takeoff  16~18 takeoff  \
0           716            788            882            833            989   
1           816            825            868            859            968   
2           977            464            366            358            186   
3          1377            575            471            388            161   
4          1682            417            361            269            241   

   18~20 takeoff bus_station_ID  
0           1028          00066  
1           1200          00075  
2            103          00031  
3            138          00024  
4            139          00012

In [ ]:
null_routes_df = test[test['route_ID'].isnull()][['bus_ID']]
print("route_ID 값이 null인 bus_ID:")
print(null_routes_df)
test.drop(['bus_ID'], axis=1, inplace=True)

route_ID 값이 null인 bus_ID:
Empty DataFrame
Columns: [bus_ID]
Index: []


In [ ]:
#고유키 생성 및 기존 칼럼 삭제
test['bus_station_PK'] = test['bus_station_ID'].astype(str) + test['bus_station_SID'].astype(str)
test.drop(['bus_station_ID'], axis=1, inplace=True)
test.drop(['bus_station_SID'], axis=1, inplace=True)
#데이터 자료형을 정수형으로 바꿈
test['bus_station_PK'] = test['bus_station_PK'].astype(int)
test.head()

month    type     route_ID  6~8 ride  8~10 ride  10~12 ride  12~14 ride  \
0  202309  서울간선버스  100100073.0       699       1292        1191         988   
1  202309  서울간선버스  123000010.0       524        838         588         627   
2  202309  서울간선버스  100100549.0       258        546         313         256   
3  202309  서울간선버스  100100009.0       295        689         420         439   
4  202309  서울간선버스  100100036.0       578        911         651         518   

   14~16 ride  16~18 ride  18~20 ride  6~8 takeoff  8~10 takeoff  \
0         744         672         720          349           716   
1         613         659         867          358           816   
2         334         482         365          681           977   
3         452         595         523          953          1377   
4         636         992         600          799          1682   

   10~12 takeoff  12~14 takeoff  14~16 takeoff  16~18 takeoff  18~20 takeoff  \
0            788            882            833            989           1028   
1            825            868            859            968           1200   
2            464            366            358            186            103   
3            575            471            388            161            138   
4            417            361            269            241            139   

   bus_station_PK  
0     66100000001  
1     75100000001  
2     31100000002  
3     24100000002  
4     12100000002

In [ ]:
test.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 113146 entries, 0 to 113145
Data columns (total 18 columns):
 #   Column          Non-Null Count   Dtype  
---  ------          --------------   -----  
 0   month           113146 non-null  int64  
 1   type            113146 non-null  object 
 2   route_ID        113146 non-null  float64
 3   6~8 ride        113146 non-null  int64  
 4   8~10 ride       113146 non-null  int64  
 5   10~12 ride      113146 non-null  int64  
 6   12~14 ride      113146 non-null  int64  
 7   14~16 ride      113146 non-null  int64  
 8   16~18 ride      113146 non-null  int64  
 9   18~20 ride      113146 non-null  int64  
 10  6~8 takeoff     113146 non-null  int64  
 11  8~10 takeoff    113146 non-null  int64  
 12  10~12 takeoff   113146 non-null  int64  
 13  12~14 takeoff   113146 non-null  int64  
 14  14~16 takeoff   113146 non-null  int64  
 15  16~18 takeoff   113146 non-null  int64  
 16  18~20 takeoff   113146 non-null  int64  
 17  bus_statio

In [ ]:
type_category = {
    '서울간선버스': 0,
    '서울지선버스': 1,
    '서울마을버스': 2,
    '서울순환버스': 3,
    '서울광역버스': 4
}
test['type'].replace(type_category, inplace=True)

test.head()

month  type     route_ID  6~8 ride  8~10 ride  10~12 ride  12~14 ride  \
0  202309     0  100100073.0       699       1292        1191         988   
1  202309     0  123000010.0       524        838         588         627   
2  202309     0  100100549.0       258        546         313         256   
3  202309     0  100100009.0       295        689         420         439   
4  202309     0  100100036.0       578        911         651         518   

   14~16 ride  16~18 ride  18~20 ride  6~8 takeoff  8~10 takeoff  \
0         744         672         720          349           716   
1         613         659         867          358           816   
2         334         482         365          681           977   
3         452         595         523          953          1377   
4         636         992         600          799          1682   

   10~12 takeoff  12~14 takeoff  14~16 takeoff  16~18 takeoff  18~20 takeoff  \
0            788            882            833            989           1028   
1            825            868            859            968           1200   
2            464            366            358            186            103   
3            575            471            388            161            138   
4            417            361            269            241            139   

   bus_station_PK  
0     66100000001  
1     75100000001  
2     31100000002  
3     24100000002  
4     12100000002

train과 test를 각각 파일로 저장한다.

In [ ]:
path = '/content/drive/MyDrive/Bus_Tayo'
train.to_csv(path+'/train.csv', index=False)

In [ ]:
path = '/content/drive/MyDrive/Bus_Tayo'
test.to_csv(path+'/test.csv', index=False)

In [ ]:
#submission을 만든다??
submission = test.copy()

#18~20 ride는 일단 0으로 초기화, 예측한 다음에 값 넣기?
submission = pd.DataFrame({'id': test['bus_station_PK'], '18~20_ride': 0})

# submission.csv 파일로 저장
submission.to_csv(path+'/submission.csv', index=False)

submission


id  18~20_ride
0        66100000001           0
1        75100000001           0
2        31100000002           0
3        24100000002           0
4        12100000002           0
...              ...         ...
113141   81999800002           0
113142    1999800003           0
113143   60999800004           0
113144  127999800005           0
113145   78999800005           0

[113146 rows x 2 columns]